<h1>Penguins Data Analysis</h1>
<hr>
<h2>Directions: Make 7 questions and analysis on the given dataset.</h2>
<h3>Analyst: Kim Andrei D. Lugatoc</h3>

In [1]:
import numpy as np
import pandas as pd

<h3>CSV file to a variable for easy call out</h3>

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h3>Adding a column named "mass_kg"</h3>

<h3>Checking for Null Values</h3>

In [30]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000, 1)

In [31]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

In [4]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h3>Filling up Null Values</|h3>

In [5]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

<h3>Using fillna() function to fill up the null values in the bill_length_mm Columns</h3>

In [6]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [7]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [8]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [9]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [10]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
dtype: int64

In [11]:
penguins['bill_depth_mm'].mean()

np.float64(17.151169590643274)

In [12]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [13]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
dtype: int64

In [14]:
penguins['flipper_length_mm'].mean()

np.float64(200.91520467836258)

In [15]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [16]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          2
sex                  0
year                 0
dtype: int64

In [17]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [18]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [19]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h3>After filling up the Null Values. Change the data types</h3>

In [20]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [21]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')

In [22]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    int64   
dtypes: category(3), float64(4), int64(1)
memory usage: 15.0 KB


<h3>Question 1: Which penguin species is the most common?</h3>

In [23]:
penguins['species'].mode()

0    Adelie
Name: species, dtype: category
Categories (3, object): ['Adelie', 'Chinstrap', 'Gentoo']

<h3>Insights: As shown above, the most common species of a penguin in this dataset is the Adelie.</h3>

<h3>Question 2: What is the heaviest penguin in the dataset?</h3>

In [32]:
penguins['mass_kg'].max()

np.float64(6.3)

In [34]:
penguins[penguins['mass_kg'] == penguins['mass_kg'].max()]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
169  Gentoo  Biscoe            49.2           15.2              221.0   

     body_mass_g   sex  year  mass_kg  
169       6300.0  male  2007      6.3

<h3>Insights: As shown above, the most heaviest penguin in this dataset is the Gentoo penguin that weighs 6.3 kilograms. </h3>

<h3>Question 3: Which island has the least penguins?</h3>

In [25]:
penguins['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h3>Insights: As shown above, the island that have the least penguins is Torgersen with only 52 penguins.</h3>

<h3>Question 4: How many penguins are male and female?</h3>

In [26]:
penguins['sex'].value_counts()

sex
male      179
female    165
Name: count, dtype: int64

<h3>Insights: As shown above, There are 179 male penguins and 165 female penguins, which leads to the total of 344.</h3>

<h3>Question 5: How many penguins weigh less than 3.5 kg?</h3>

In [38]:
penguins[penguins['mass_kg'] < 3.5]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
2       Adelie  Torgersen            40.3           18.0              195.0   
4       Adelie  Torgersen            36.7           19.3              193.0   
10      Adelie  Torgersen            37.8           17.1              186.0   
12      Adelie  Torgersen            41.1           17.6              182.0   
16      Adelie  Torgersen            38.7           19.0              195.0   
..         ...        ...             ...            ...                ...   
326  Chinstrap      Dream            48.1           16.4              199.0   
330  Chinstrap      Dream            42.5           17.3              187.0   
331  Chinstrap      Dream            52.2           18.8              197.0   
332  Chinstrap      Dream            45.2           16.6              191.0   
340  Chinstrap      Dream            43.5           18.1              202.0   

     body_mass_g     sex  year  mass_kg  
2         3250.0  female  2007      3.2  
4         3450.0  female  2007      3.4  
10        3300.0    male  2007      3.3  
12        3200.0  female  2007      3.2  
16        3450.0  female  2007      3.4  
..           ...     ...   ...      ...  
326       3325.0  female  2009      3.3  
330       3350.0  female  2009      3.4  
331       3450.0    male  2009      3.4  
332       3250.0  female  2009      3.2  
340       3400.0  female  2009      3.4  

[68 rows x 9 columns]

<h3>Insights: As shown in the table above, there are 68 penguins that weight less than 3.5kg.</h3>

<h3>Question 6: What is the maximum flipper length among penguins heavier than 4 kg?</h3>

In [47]:
penguins[penguins['mass_kg'] < 4]['flipper_length_mm'].max()

np.float64(202.0)

<h3>Insights: The maximum flipper length of a penguin that is heavier than 4 kilograms is 202mm.</h3>

<h3>Questions 7: Which penguin is the lightest in Torgersen island?</h3>

In [52]:
min_weight = penguins[penguins['island'] == 'Torgersen']['mass_kg'].min()
penguins[(penguins['island'] == 'Torgersen') & (penguins['mass_kg'] == min_weight)]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
116  Adelie  Torgersen            38.6           17.0              188.0   

     body_mass_g     sex  year  mass_kg  
116       2900.0  female  2009      2.9

<h3>Insights: As shown above the data frame, The lightest penguin in Torgersen is an Adelie penguin and she weighs 2.9 kilograms</h3>